### DISCUSSION TITLE
Article: When smart algorithms beat artificial intelligence -brute force

### DISCUSSION THREAD

**Discussion by Gurigraphics** [ggml-org/whisper.cpp#2018] [Show and tell]
It can be useful for those looking to improve transcription quality
Mainly for voice assistant executing "known commands", such as do this or that
And even more so when memory and processing resources are scarce: embedded, IoT, etc

https://dev.to/gurigraphics/when-smart-algorithms-beat-ai-brute-force-57mj

**Comment by liam-mceneaney** [comment]
Good article, the wchess example actually already uses levenshtein distance!  https://github.com/ggerganov/whisper.cpp/blob/8f253ef3af1c62c04316ba4afa7145fc4d701a8c/examples/wchess/libwchess/WChess.cpp#L74

https://github.com/ggerganov/whisper.cpp/blob/8f253ef3af1c62c04316ba4afa7145fc4d701a8c/examples/common.cpp#L786

**Reply by Gurigraphics** [reply]
Thanks for the complement
Devs who are not from this area generally don't even know that it exists or what it is for
In the article I also mentioned others: Jaro–Winkler Similarity, Jaccard Similarity

Another possibility is to use "embeddings" for semantic similarity
In this case the model would be able to recognize that "Brazil" is a "Country" and "orange" is a "fruit" or "color"
